admin_guide/cluster_metrics.adoc

```diff
@@ -16,298 +16,382 @@ As an OpenShift administrator, you may want to view the metrics from all
+
+The following command will give the heapster service account this required permission:
+
+[options="nowrap"]
 ----
-$ oc scale rc influxdb-monitoring --replicas=<number>
+$ oadm policy add-cluster-role-to-user cluster-reader system:serviceaccount:metrics:heapster
 ----
 
-== Deploying Heapster
+== Metric Data Storage
 
-*Heapster* reads all nodes and pods from the master, then connects to each
-node's kubelet to retrieve pod metrics. *Heapster* will run under a
-link:service_accounts.html[service account] with the *cluster-reader*
-link:manage_authorization_policy.html#viewing-cluster-policy[cluster role] to
-be able to retrieve these metrics.
+The metrics data can be stored to either 
+link:../architecture/additional_concepts/storage.html[persistent storage] or to a
+temporary link:../dev_guide/volumes.html[pod volume].
 
-The following is the definition of a service account for *Heapster* to use:
+=== Persistent Storage
 
-====
-[source,yaml]
-----
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: heapster
-----
-====
+Running with persistent storage means that your metrics will be stored to a 
+link:../architecture/additional_concepts/storage.html#persistent-volumes[persistent volume]
+and be able to survive a pod being restarted or recreated. Running with persistent storage
+is highly recommended if you require your metric data to be guarded from data loss.
+
+In order to use persistent storage with the metric components, all that is required is that a
+link:../architecture/additional_concepts/storage.html#persistent-volumes[persistent volume]
+of sufficient size is available.
+
+The creation of 
+link:../architecture/additional_concepts/storage.html#persistent-volume-claims[persisten volume claims]
+is handled by the metrics deployer.
+
+
+=== Non-Persistent Storage
+
+Running with non-persistent storage means that any stored metrics will be deleted with the pod. 
+Metrics will still survive a container being restarted. It is 
+much easier to run with non-persistent data, but with the tradeoff of potentially losing 
+this metric data. Running with non-persistent data should only be done when data loss is acceptable.
+
+In order to use non-persistent storage, you will need to set the 
+`USE_PERSISTENT_STORAGE` template option to false for the deployer.
+
+
+== Metrics Deployer
+
+The `metrics deployer` is what is used to deploy and configure all the various metrics
+components.
 
-Save it to a file and create it with:
+The `metrics deployer` can configured by passing in information from
+link:../dev_guide/secrets.html[secrets] or by passing parameters to the deployer's 
+link:../dev_guide/templates.html[template].
 
+=== Metrics Deployer Secrets
+
+By default the deployer will autogenerate self signed certificates to be used between the various
+components. If you wish to provide your own certificates, it is possible to do so by passing these
+values as link:../dev_guide/secrets.html[secrets] to the deployer.
```

Comment:
We should emphasize that the certificates must be trusted by the browser. Otherwise metrics will not show up in the web console.